src/Pool.sol

```diff
@@ -178,14 +178,14 @@ abstract contract Pool {
         return uint128(collected);
     }
 
-    /// @notice Collects all received funds available for collection
-    /// by the sender of the message and sends them to that sender
-    function collect() public virtual {
-        uint128 collected = _collectInternal(msg.sender);
+    /// @notice Collects all received funds available for the user and sends them to that user
+    /// @param id The id of the user.
+    function collect(address id) public virtual {
```

Comment:
This method would allow an attack on the current NFT design.
We will change it anyway, so it doesn't matter.

However, together with the current NFT based approach I could just calculate your NFT `address id` and call collect.
Your funds would be sent to the `address id` instead of msg.sender.

Reply:
Hmm, yes and no. You're right, if one decides to send funds to an ID calculated for an NFT, then the funds can be locked by a malicious 3rd party. But AFAIK streaming to an NFT is not a supported use case. If one wants to stream to an arbitrary address, we can't guarantee that it'll be collectable by anybody.

Reply:
I added this method: https://github.com/radicle-dev/radicle-streaming/blob/master/src/NFTPool.sol#L32

Maybe you can remove it. Then I am fine with merging now.

We anyway will implement the streaming NFT proposal.

Reply:
Removed